services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/service/Impl/SegmentServiceImpl.java

```diff
@@ -0,0 +1,254 @@
+        }
+    }
+
+    @Override
+    public VlanKeyRequest allocateVlan(VlanKeyRequest request) throws Exception {
+        logger.debug("Allocate vlan key, request: {}", request);
+
+        Long key = vlanRangeRepository.allocateVlanKey(request.getRangeId());
+
+        request.setKey(key);
+
+        logger.info("Allocate vlan key success, request: {}", request);
+
+        return request;
+    }
+
+    @Override
+    public VlanKeyRequest releaseVlan(String networkType, String rangeId, Long key) throws Exception {
+        logger.debug("Release vlan key, ipAddr: {}", key);
+
+        vlanRangeRepository.releaseVlanKey(rangeId, key);
+
+        VlanKeyRequest result = new VlanKeyRequest();
+        result.setNetworkType(networkType);
+        result.setRangeId(rangeId);
+        result.setKey(key);
+
+        logger.info("Release vlan key success, result: {}", result);
+
+        return result;
+    }
+
+    @Override
+    public VlanKeyRequest getVlan(String networkType, String rangeId, Long key) throws Exception {
+        logger.debug("Get vlan key, rangeId: {}, ipAddr: {}", rangeId, key);
+
+        VlanKeyAlloc vlanKeyAlloc = vlanRangeRepository.getVlanKeyAlloc(rangeId, key);
+        if (vlanKeyAlloc.getNetworkType() != networkType) {
+            throw new NetworkTypeInvalidException();
+        }
+
+        VlanKeyRequest result = new VlanKeyRequest();
+
+        result.setNetworkType(vlanKeyAlloc.getNetworkType());
+        result.setRangeId(vlanKeyAlloc.getRangeId());
+        result.setKey(vlanKeyAlloc.getKey());
+
+        logger.info("Get vlan key success, result: {}", result);
+
+        return result;
+    }
+
+    @Override
+    public NetworkVlanRangeRequest createRange(NetworkVlanRangeRequest request) throws Exception {
+        logger.debug("Create vlan range, request: {}", request);
+
+        vlanRangeRepository.createRange(request);
+
+        logger.info("Create vlan range success, request: {}", request);
+
+        return request;
+    }
+
+    @Override
+    public NetworkVlanRangeRequest deleteRange(String rangeId) throws Exception {
+        logger.debug("Delete vlan range, rangeId: {}", rangeId);
+
+        NetworkVlanRange networkVlanRange = vlanRangeRepository.deleteRange(rangeId);
+
+        NetworkVlanRangeRequest request = new NetworkVlanRangeRequest();
+        request.setId(networkVlanRange.getId());
+        request.setSegmentId(networkVlanRange.getSegmentId());
+        request.setNetworkType(networkVlanRange.getNetworkType());
+        request.setFirstKey(networkVlanRange.getFirstKey());
+        request.setLastKey(networkVlanRange.getLastKey());
+        request.setUsedKeys(networkVlanRange.getUsedKeys());
+        request.setTotalKeys(networkVlanRange.getTotalKeys());
+
+        logger.info("Delete vlan range success, request: {}", request);
+
+        return request;
+    }
+
+    @Override
+    public NetworkVlanRangeRequest getRange(String rangeId) throws Exception {
```

Comment:
Either rename the method to GetVlanRange, or change the response type to a more generic name like NetworkSegmentRange etc..

Reply:
Okay ~ you are right